src/Validation/gen/ValidationsGenerator.cs

```diff
@@ -11,28 +14,70 @@ public sealed partial class ValidationsGenerator : IIncrementalGenerator
 {
     public void Initialize(IncrementalGeneratorInitializationContext context)
     {
+        // Get the MSBuild properties for controlling attribute generation
+        var generatorSettings = context.AnalyzerConfigOptionsProvider
+            .Select((configOptions, _) =>
+            {
+                configOptions.GlobalOptions.TryGetValue("build_property.RootNamespace", out var rootNamespace);
+                configOptions.GlobalOptions.TryGetValue("build_property.GenerateBuiltinValidatableTypeAttribute", out var generateAttribute);
+
+                var shouldGenerate = !string.IsNullOrEmpty(rootNamespace) &&
+                                   string.Equals(generateAttribute, "true", StringComparison.OrdinalIgnoreCase);
+
+                return new GeneratorSettings(rootNamespace, shouldGenerate);
+            });
+
+        // Emit the ValidatableTypeAttribute conditionally
+        context.RegisterSourceOutput(generatorSettings, EmitValidatableTypeAttribute);
+
         // Find the builder.Services.AddValidation() call in the application.
         var addValidation = context.SyntaxProvider.CreateSyntaxProvider(
             predicate: FindAddValidation,
             transform: TransformAddValidation
         );
+
         // Extract types that have been marked with [ValidatableType].
+        // This handles both the framework attribute and the auto-generated attribute.
         var validatableTypesWithAttribute = context.SyntaxProvider.ForAttributeWithMetadataName(
             "Microsoft.Extensions.Validation.ValidatableTypeAttribute",
             predicate: ShouldTransformSymbolWithAttribute,
             transform: TransformValidatableTypeWithAttribute
         );
+
+        // Extract types that have been marked with the embedded [ValidatableType].
+        var generatedValidatableTypesWithAttribute = context.SyntaxProvider.CreateSyntaxProvider(
+            predicate: ShouldTransformSymbolWithEmbeddedValidatableTypeAttribute,
+            transform: (context, cancellation) => context
+        ).Combine(generatorSettings)
+            .Select((combined, cancellation) =>
+            {
+                var (syntaxContext, settings) = combined;
+                if (!settings.ShouldGenerateAttribute || string.IsNullOrEmpty(settings.RootNamespace))
+                {
+                    return [];
+                }
+
+                return TransformValidatableTypeWithEmbeddedValidatableTypeAttribute(syntaxContext, settings, cancellation);
+            })
+            .Where(t => t.Length > 0);
+
+        var allValidatableTypes = generatedValidatableTypesWithAttribute.Collect()
+            .Combine(validatableTypesWithAttribute.Collect())
+            .SelectMany((pair, cancellation) => pair.Left.Concat(pair.Right).ToImmutableArray());
```

Comment:
This I'm not sure it's the best way to do it. I believe everytime we call "Collect" we are grouping stuff caching wise, so we need to think this through